deps/rabbit/src/rabbit_node_monitor.erl

```diff
@@ -323,20 +326,53 @@ find_blocked_global_peers1([], _) ->
 
 unblock_global_peer(PeerNode) ->
     ThisNode = node(),
-    PeerState = rpc:call(PeerNode, sys, get_status, [global_name_server]),
+
+    {status, _, _, [_, _, _, _, PeerState]} = PeerStatus =
+        erpc:call(PeerNode, sys, get_status, [global_name_server]),
+    {status, _, _, [_, _, _, _, ThisState]} = ThisStatus =
+        sys:get_status(global_name_server),
```

Comment:
You should use `sys:get_state/1` now to get the state directly, without having to parse the return value of `sys:get_status/1`.

Reply:
I first used `sys:get_state/1`, but moved to `sys:get_status/1` because the docs for `sys:get_state/1` say:
>These functions are intended only to help with debugging. They are provided for convenience, allowing developers to avoid having to create their own state extraction functions and also avoid having to interactively extract the state from the return values of get_status/1 or get_status/2 **while debugging**.

But I can move back to  `sys:get_state/1` if you think it's safe to use in production?

Reply:
I changed it back to `sys:get_state/1`.